sdk/servicebus/Azure.Messaging.ServiceBus/tests/RuleManager/RuleManagerLiveTests.cs

```diff
@@ -0,0 +1,589 @@
+                    {
+                        CorrelationId = "correlationId",
+                        Subject = "label",
+                        MessageId = "messageId",
+                        ApplicationProperties =
+                            {
+                                {"key1", "value1"}
+                            },
+                        ReplyTo = "replyTo",
+                        ReplyToSessionId = "replyToSessionId",
+                        SessionId = "sessionId",
+                        To = "to"
+                    },
+                    Action = new SqlRuleAction("Set CorrelationId = 'newValue'")
+                };
+                await ruleManager.AddRuleAsync(ruleOptions);
+
+                rules = (await ruleManager.GetRulesAsync()).ToList();
+                Assert.AreEqual(3, rules.Count);
+
+                var sqlRule = rules.FirstOrDefault(rule => rule.Name.Equals(sqlRuleName));
+                Assert.NotNull(sqlRule);
+                Assert.Null(sqlRule.Action);
+                Assert.IsInstanceOf<SqlRuleFilter>(sqlRule.Filter);
+                Assert.AreEqual("price > 10", ((SqlRuleFilter)sqlRule.Filter).SqlExpression);
+
+                var correlationRule = rules.FirstOrDefault(rule => rule.Name.Equals(correlationRuleName));
+                Assert.NotNull(correlationRule);
+                Assert.IsInstanceOf<SqlRuleAction>(correlationRule.Action);
+                var sqlRuleAction = correlationRule.Action as SqlRuleAction;
+                Assert.NotNull(sqlRuleAction);
+                Assert.AreEqual("Set CorrelationId = 'newValue'", sqlRuleAction.SqlExpression);
+                Assert.IsInstanceOf<CorrelationRuleFilter>(correlationRule.Filter);
+                var correlationRuleFilter = correlationRule.Filter as CorrelationRuleFilter;
+                Assert.NotNull(correlationRuleFilter);
+                Assert.AreEqual("correlationId", correlationRuleFilter.CorrelationId);
+                Assert.AreEqual("label", correlationRuleFilter.Subject);
+                Assert.AreEqual("messageId", correlationRuleFilter.MessageId);
+                Assert.AreEqual("replyTo", correlationRuleFilter.ReplyTo);
+                Assert.AreEqual("replyToSessionId", correlationRuleFilter.ReplyToSessionId);
+                Assert.AreEqual("sessionId", correlationRuleFilter.SessionId);
+                Assert.AreEqual("to", correlationRuleFilter.To);
+                Assert.NotNull(correlationRuleFilter.ApplicationProperties);
+                Assert.AreEqual("value1", correlationRuleFilter.ApplicationProperties["key1"]);
+
+                await ruleManager.RemoveRuleAsync(RuleProperties.DefaultRuleName);
+                await ruleManager.RemoveRuleAsync(sqlRuleName);
+                await ruleManager.RemoveRuleAsync(correlationRuleName);
+                rules = (await ruleManager.GetRulesAsync()).ToList();
+                Assert.AreEqual(0, rules.Count());
```

Comment:
```suggestion
                Assert.AreEqual(0, rules.Count);
```

Not impactful, but we know we have a list here.